src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -757,6 +771,9 @@ def open(self):
             if self.driver is not None:
                 kwargs.update(driver=self.driver)
 
+                if self.driver == "ros3" and self.aws_region is not None:
+                    kwargs.update(aws_region=bytes(self.aws_region, "ascii"))
```

Comment:
Hmmm, again not sure how to test this `open` method, I don't think I ever use that with respect to this class

Deferring to y'all to add a test

Reply:
`open` is called in here

https://github.com/hdmf-dev/hdmf/blob/f163a768dc33409df5014f4c4e487f1713c7d61c/src/hdmf/backends/io.py#L41

in the parent `HDMFIO.__init__`. However, it looks like the `awsregion` etc. is not being passed through:

https://github.com/hdmf-dev/hdmf/blob/f163a768dc33409df5014f4c4e487f1713c7d61c/src/hdmf/backends/hdf5/h5tools.py#L104

So, normally, your `test_basic_read_with_aws_region` should excercise  your new code in `open`  method. The fact that it does not get hit seems suspicious. I.e., I would add a breackpoint in line 772 to check what `self.dirver` and `self.aws_region` are set to when you run that test.